java/src/main/java/com/logicalclocks/hsfs/engine/HudiEngine.java

```diff
@@ -165,8 +166,11 @@ private FeatureGroupCommit getLastCommitMetadata(SparkSession sparkSession, Stri
     hudiArgs.put(HUDI_WRITE_INSERT_DROP_DUPLICATES, "true");
 
     // primary keys
-    Seq<String> primaryColumns = utils.getPrimaryColumns(featureGroup);
-    hudiArgs.put(HUDI_RECORD_KEY, primaryColumns.mkString(","));
+    String primaryColumns = utils.getPrimaryColumns(featureGroup).mkString(",");
+    if (!Strings.isNullOrEmpty(featureGroup.getEventTime())) {
+      primaryColumns = primaryColumns + "," + featureGroup.getEventTime();
```

Comment:
This looks a bit weird, why not put it in the collection just have a single string manipulation?

Reply:
I agree, and I am sure I tried, but the collection is a Scala Seq, and I could not manage adding an element using scala syntax from java.